### ISSUE TITLE
Eval bug: Qwen3.6-27B garbled output on dual-GPU CUDA (RTX 5060 Ti + RTX 3060), single GPU works

### ISSUE THREAD

**Issue by codesftdev** [ggml-org/llama.cpp#26257]
### Name and Version

llama-cli --version
version: 10167 (388e17d)
built with GNU 16.1.1 for Linux x86_64

### Operating systems

Linux

### GGML backends

CUDA

### Hardware

Ryzen 5 5600X
RTX 5060 Ti 16GB
RTX 3060 12GB

### Models

Qwen 3.6 27B [(Unsloth UD-Q5_K_M)](https://huggingface.co/unsloth/Qwen3.6-27B-MTP-GGUF?show_file_info=Qwen3.6-27B-Q5_K_M.gguf)

### Problem description & steps to reproduce

Running Qwen 3.6 27B on the RTX 3060 12GB and the RTX 5060 Ti 16GB results in garbled output, but running on the RTX 5060 Ti + system RAM does not result in garbled output. This occured using llama-serve, llama-cli, and llama-complete.

Both GPUs: `llama-completion -hf unsloth/Qwen3.6-27B-MTP-GGUF:UD-Q5_K_XL -fit off -c 4096 -n 50 --temp 0 --seed 42 -p "hello"`
Single GPU + System RAM: ` CUDA_VISIBLE_DEVICES=0 llama-completion -hf unsloth/Qwen3.6-27B-MTP-GGUF:UD-Q5_K_XL -c 4096 --verbose -n 50 --temp 0 --seed 42 -p "hello"`

### Relevant log output

<details>
<summary>Logs</summary>

```console
❯ llama-completion \
        -hf unsloth/Qwen3.6-27B-MTP-GGUF:UD-Q5_K_XL \
        -fit off \
        -c 4096 \
        -n 50 \
        --temp 0 \
        --seed 42 \
        -p "hello"
0.00.467.889 I llama_completion: llama backend init
0.00.467.894 I llama_completion: load the model and apply lora adapter, if any
0.11.621.054 I llama_completion: llama threadpool init, n_threads = 6
0.11.621.066 I llama_completion: chat template is available, enabling conversation mode (disable it with -no-cnv)
0.11.621.066 W *** User-specified prompt will pre-start conversation, did you mean to set --system-prompt (-sys) instead?
0.11.627.053 I llama_completion: chat template example:
<|im_start|>system
You are a helpful assistant<|im_end|>
<|im_start|>user
Hello<|im_end|>
<|im_start|>assistant
Hi there<|im_end|>
<|im_start|>user
How are you?<|im_end|>
<|im_start|>assistant

0.11.627.055 I
0.11.627.094 I system_info: n_threads = 6 (n_threads_batch = 6) / 12 | CUDA : ARCHS = 860,1200 | USE_GRAPHS = 1 | FA_ALL_QUANTS = 1 | BLACKWELL_NATIVE_FP4 = 1 | CPU : SSE3 = 1 | SSSE3 = 1 | AVX = 1 | AVX2 = 1 | F16C = 1 | FMA = 1 | BMI2 = 1 | LLAMAFILE = 1 | OPENMP = 1 | REPACK = 1 |
0.11.627.095 I
0.11.627.195 I llama_completion: interactive mode on.
0.11.627.198 I sampler seed: 42
0.11.627.209 I sampler params:
        repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
        dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
        top_k = 20, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.000
        mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000, adaptive_target = -1.000, adaptive_decay = 0.900
0.11.627.213 I sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist
0.11.627.214 I generate: n_ctx = 4096, n_batch = 2048, n_predict = 50, n_keep = 0
0.11.627.214 I
0.11.627.214 I == Running in interactive mode. ==
0.11.627.214 I  - Press Ctrl+C to interject at any time.
0.11.627.215 I  - Press Return to return control to the AI.
 - To return control without starting a new line, end your input with '/'.
 - If you want to submit another line, end your input with '\'.
0.11.627.215 I  - Not using system message. To change it, set a different value via -sys PROMPT
0.11.627.215 I
user
hello
assistant
保证保证保证巻き巻き保证保证

<think>保证保证

保证保证巻きsWith保证保证保证绿城巻き保证保证保证廷巻き保证



保证保证巻き巻き

保证巻きsWith

保证保证巻き保证保证sWithsWithsWith

保证

保证
>
0.25.041.089 I common_perf_print:    sampling time =      19.33 ms
0.25.041.092 I common_perf_print:    samplers time =       7.21 ms /    59 tokens
0.25.041.096 I common_perf_print:        load time =   10726.31 ms
0.25.041.097 I common_perf_print: prompt eval time =     484.59 ms /     9 tokens (   53.84 ms per token,    18.57 tokens per second)
0.25.041.098 I common_perf_print:        eval time =    2826.48 ms /    49 runs   (   57.68 ms per token,    17.34 tokens per second)
0.25.041.099 I common_perf_print:       total time =   13420.04 ms /    58 tokens
0.25.041.100 I common_perf_print: unaccounted time =   10089.64 ms /  75.2 %      (total - sampling - prompt eval - eval) / (total)
0.25.041.100 I common_perf_print:    graphs reused =         49
Interrupted by user
```

```console
❯ CUDA_VISIBLE_DEVICES=0 llama-completion \
        -hf unsloth/Qwen3.6-27B-MTP-GGUF:UD-Q5_K_XL \
        -c 4096 \
        -n 50 \
        --temp 0 \
        --seed 42 \
        -p "hello"
0.00.626.413 I llama_completion: llama backend init
0.00.626.420 I llama_completion: load the model and apply lora adapter, if any
0.33.715.705 I llama_completion: llama threadpool init, n_threads = 6
0.33.715.717 I llama_completion: chat template is available, enabling conversation mode (disable it with -no-cnv)
0.33.715.717 W *** User-specified prompt will pre-start conversation, did you mean to set --system-prompt (-sys) instead?
0.33.715.832 I llama_completion: chat template example:
<|im_start|>system
You are a helpful assistant<|im_end|>
<|im_start|>user
Hello<|im_end|>
<|im_start|>assistant
Hi there<|im_end|>
<|im_start|>user
How are you?<|im_end|>
<|im_start|>assistant

0.33.715.834 I
0.33.715.865 I system_info: n_threads = 6 (n_threads_batch = 6) / 12 | CUDA : ARCHS = 860,1200 | USE_GRAPHS = 1 | FA_ALL_QUANTS = 1 | BLACKWELL_NATIVE_FP4 = 1 | CPU : SSE3 = 1 | SSSE3 = 1 | AVX = 1 | AVX2 = 1 | F16C = 1 | FMA = 1 | BMI2 = 1 | LLAMAFILE = 1 | OPENMP = 1 | REPACK = 1 |
0.33.715.866 I
0.33.715.936 I llama_completion: interactive mode on.
0.33.715.938 I sampler seed: 42
0.33.715.950 I sampler params:
        repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
        dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
        top_k = 20, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.000
        mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000, adaptive_target = -1.000, adaptive_decay = 0.900
0.33.715.954 I sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist
0.33.715.955 I generate: n_ctx = 4096, n_batch = 2048, n_predict = 50, n_keep = 0
0.33.715.957 I
0.33.715.957 I == Running in interactive mode. ==
0.33.715.957 I  - Press Ctrl+C to interject at any time.
0.33.715.958 I  - Press Return to return control to the AI.
 - To return control without starting a new line, end your input with '/'.
 - If you want to submit another line, end your input with '\'.
0.33.715.958 I  - Not using system message. To change it, set a different value via -sys PROMPT
0.33.715.958 I
user
hello
assistant
<think>

</think>

                                                                                                                                                                                       Hello! How can I help you today?

>
0.40.577.575 I common_perf_print:    sampling time =       6.53 ms
0.40.577.576 I common_perf_print:    samplers time =       2.02 ms /    23 tokens
0.40.577.582 I common_perf_print:        load time =   31073.74 ms
0.40.577.583 I common_perf_print: prompt eval time =     965.59 ms /     9 tokens (  107.29 ms per token,     9.32 tokens per second)
0.40.577.585 I common_perf_print:        eval time =    3894.63 ms /    13 runs   (  299.59 ms per token,     3.34 tokens per second)
0.40.577.586 I common_perf_print:       total time =    6861.88 ms /    22 tokens
0.40.577.586 I common_perf_print: unaccounted time =    1995.12 ms /  29.1 %      (total - sampling - prompt eval - eval) / (total)
0.40.577.587 I common_perf_print:    graphs reused =         12
Interrupted by user
```

Verbose Single-GPU + System RAM logs available at [https://gist.github.com/codesftdev/cb34619f29a9130a9604512d5081fbd5](https://gist.github.com/codesftdev/cb34619f29a9130a9604512d5081fbd5)
</details>

**Comment by DevenBL** [comment]
I experienced the same thing with a different quant of Qwen3.6-27B.  It seems to be a back end agnostic bug. Affecting at the very least both the cuda and vulkan backends.

**Comment by codesftdev** [comment]
llama.cpp built for Vulkan (b10181-caa596ab3) seems to have the same issue as well.

```console
❯ ./build/bin/llama-completion -hf unsloth/Qwen3.6-27B-MTP-GGUF:UD-Q5_K_XL -fit off -c 4096 -n 50 --temp 0 --seed 42 -p "hello"

0.00.300.412 I llama_completion: llama backend init
0.00.300.422 I llama_completion: load the model and apply lora adapter, if any
0.11.875.656 I llama_completion: llama threadpool init, n_threads = 6
0.11.875.676 I llama_completion: chat template is available, enabling conversation mode (disable it with -no-cnv)
0.11.875.677 W *** User-specified prompt will pre-start conversation, did you mean to set --system-prompt (-sys) instead?
0.11.875.775 I llama_completion: chat template example:
<|im_start|>system
You are a helpful assistant<|im_end|>
<|im_start|>user
Hello<|im_end|>
<|im_start|>assistant
Hi there<|im_end|>
<|im_start|>user
How are you?<|im_end|>
<|im_start|>assistant

0.11.875.777 I
0.11.875.824 I system_info: n_threads = 6 (n_threads_batch = 6) / 12 | CPU : SSE3 = 1 | SSSE3 = 1 | AVX = 1 | AVX2 = 1 | F16C = 1 | FMA = 1 | BMI2 = 1 | LLAMAFILE = 1 | OPENMP = 1 | REPACK = 1 |
0.11.875.826 I
0.11.875.901 I llama_completion: interactive mode on.
0.11.875.904 I sampler seed: 42
0.11.875.912 I sampler params:
        repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
        dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
        top_k = 20, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.000
        mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000, adaptive_target = -1.000, adaptive_decay = 0.900
0.11.875.918 I sampler chain: logits -> ?penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist
0.11.875.918 I generate: n_ctx = 4096, n_batch = 2048, n_predict = 50, n_keep = 0
0.11.875.918 I
0.11.875.918 I == Running in interactive mode. ==
0.11.875.919 I  - Press Ctrl+C to interject at any time.
0.11.875.919 I  - Press Return to return control to the AI.
 - To return control without starting a new line, end your input with '/'.
 - If you want to submit another line, end your input with '\'.
0.11.875.919 I  - Not using system message. To change it, set a different value via -sys PROMPT
0.11.875.919 I
user
hello
assistant
anness，anness <think>
，anness响度诸
</think>，anness,响响
响，AHAN响
响，，响响响响响响响响响响响响响响，响响响响响响响响
>
0.35.758.199 I common_perf_print:    sampling time =      14.30 ms
0.35.758.200 I common_perf_print:    samplers time =       5.25 ms /    59 tokens
0.35.758.204 I common_perf_print:        load time =   11490.78 ms
0.35.758.205 I common_perf_print: prompt eval time =     193.35 ms /     9 tokens (   21.48 ms per token,    46.55 tokens per second)
0.35.758.206 I common_perf_print:        eval time =    2778.15 ms /    49 runs   (   56.70 ms per token,    17.64 tokens per second)
0.35.758.206 I common_perf_print:       total time =   23882.55 ms /    58 tokens
0.35.758.207 I common_perf_print: unaccounted time =   20896.75 ms /  87.5 %      (total - sampling - prompt eval - eval) / (total)
0.35.758.207 I common_perf_print:    graphs reused =         49
Interrupted by user```

**Comment by DevenBL** [comment]
This also occurs for Qwen3.8-27B

**Comment by codesftdev** [comment]
> This also occurs for Qwen3.8-27B

For some reason, I'm getting normal output from Qwen3.8-27B across both GPUs, using a CUDA build of b10456-f17008d. It worked with unsloth/Qwen3.8-27B-GGUF (UD-Q5_K_XL) and empero-ai/Qwen3.8-27B-Ridge-GGUF

**Comment by DevenBL** [comment]
> > This also occurs for Qwen3.8-27B
> 
> For some reason, I'm getting normal output from Qwen3.8-27B across both GPUs, using a CUDA build of b10456-f17008d. It worked with unsloth/Qwen3.8-27B-GGUF (UD-Q5_K_XL) and empero-ai/Qwen3.8-27B-Ridge-GGUF

Perhaps it's isolated to vulkan only now? These are the options i used to load it. It's an unsloth quant,

```
[qwen3.8-27b-duo]
# --- base model ---
model = /home/dev/Git/models/Qwen3.8-27B-Q4_K_S.gguf
device = Vulkan0,Vulkan1
jinja = true
chat-template-file = /home/dev/Git/models/Qwen3.8-chat_template.jinja

# --- compute / GPU ---
flash-attn = on
cache-type-k = q4_0
cache-type-v = q4_0
tensor-split = 8,10
no-perf = true
no-kv-offload = true
swa-full = true
n-gpu-layers = 99         # was -ngl 7

# --- context ---
ctx-size = 0             # native ctx

# --- reasoning ---
reasoning = auto
reasoning-preserve = true

# --- draft / speculative ---
# MTP draft is integrated into the model — no external model-draft path needed
cache-type-k-draft  = q4_0
cache-type-v-draft  = q4_0
device-draft = Vulkan0
spec-type = draft-mtp
```

version: 0.1.2-dev (build 10517, commit dc72703fc)

**Comment by DevenBL** [comment]
Fixed on Vulkan for v0.5.0
